### PR TITLE
Update code so it can use OpenCV 3 libs

### PR DESCRIPTION
**Pull request by TSC21**
@bloesch, I updated the code so it can use OpenCV 3 libs/binaries. It already compiles but I get a seg fault when running any of the nodes. Using `gdb`, I get:
```bash
Program received signal SIGSEGV, Segmentation fault.
0x00007ffff6bb4282 in cv::Mat::copyTo(cv::_OutputArray const&) const () from /usr/lib/x86_64-linux-gnu/libopencv_core.so.2.4
```
The above may be related to the fact that `libopencv-*2.4` is installed, but removing it breaks ROS install. If anyone has a recommendation on how to solve the symbolic link problem, I appreciate it, and everyone will benefit. It may be just a little detail in `CMakeLists.txt`, which I'm missing.

Recommended: install OpenCV 3 by src. Seems like that the build doesn't link to `ros-<ros-distro>-opencv3`. If I remove `find_package(OpenCV REQUIRED)` and `target_link_libraries(${PROJECT_NAME} ... ${OpenCV_LIBS}`, it gets symbolic link errors during compile time.

### CODE REVIEW

**Comment by mhkabir** [comment]
You  necessarily need to install OpenCV3 from source. The ROS OpenCV3 metapackages don't work properly. Keeping it side by side with CV2.4 isn't a good idea either. 

I compiled CV3 from source and compiled the dependent ROS packages from source.

File: CMakeLists.txt [hunk 1]
```diff
@@ -20,8 +20,19 @@ if(0)
 	find_package(OpenMP REQUIRED)
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
 endif()
+
+find_package(OpenCV 3.1.0 REQUIRED)
```
Comment (mhkabir) [inline CMakeLists.txt]: This will apparently break 2.4 builds.

Comment (TSC21) [inline CMakeLists.txt]: If you don't use this, it won't build against Opencv 3.

**Comment by TSC21** [comment]
> I compiled CV3 from source and compiled the dependent ROS packages from source.

I've done the same earlier and it breaks. Have you tried this branch already and see how it goes?

**Comment by TSC21** [comment]
> The ROS OpenCV3 metapackages don't work properly

This is not entirely true, since I have other ROS apps running against those and it works perfectly fine.

**Comment by TSC21** [comment]
> Keeping it side by side with CV2.4 isn't a good idea either.

I didn't install 2.4 from source. It installs automatically (`libopencv-*2.4`) when installing some of the ROS packages that come with `desktop-full`.

**Comment by TSC21** [comment]
OK I was able to solve this by adding `vision_opencv` src to the ws and build all together. Functioning right now.

**Comment by bloesch** [comment]
Closing - solved in another PR.

**Pull request closed without merging**